components/match2/wikis/criticalops/match_group_input_custom.lua

```diff
@@ -0,0 +1,596 @@
+	return Table.all(table, function (_, scoreinfo) return scoreinfo.status == STATUS_SCORE end)
+end
+
+--
+-- match related functions
+--
+function matchFunctions.getBestOf(match)
+	local mapCount = 0
+	for _, _, mapIndex in Table.iter.pairsByPrefix(match, 'map') do
+		mapCount = mapIndex
+	end
+	match.bestof = mapCount
+	return match
+end
+
+-- Template:Map sets a default map name so we can count the number of maps.
+-- These maps however shouldn't be stored in lpdb, nor displayed
+-- The discardMap function will check if a map should be removed
+-- Remove all maps that should be removed.
+function matchFunctions.removeUnsetMaps(match)
+	for mapKey, map in Table.iter.pairsByPrefix(match, 'map') do
+			if map.map == DUMMY_MAP_NAME then
+				match[mapKey] = nil
+			end
+	end
+	return match
+end
+
+-- Calculate the match scores based on the map results.
+-- If it's a Best of 1, we'll take the exact score of that map
+-- If it's not a Best of 1, we should count the map wins
+-- Only update a teams result if it's
+-- 1) Not manually added
+-- 2) At least one map has a winner
+function matchFunctions.getScoreFromMapWinners(match)
+	-- For best of 1, display the results of the single map
+	local opponent1 = match.opponent1
+	local opponent2 = match.opponent2
+	local newScores = {}
+	local foundScores = false
+	if match.bestof == 1 then
+		if match.map1 then
+			newScores = match.map1.scores
+			foundScores = true
+		end
+	else -- For best of >1, disply the map wins
+		for _, map in Table.iter.pairsByPrefix(match, 'map') do
+				local winner = tonumber(map.winner)
+				foundScores = true
+			-- Only two opponents in C-OPS
+				if winner and winner > 0 and winner <= 2 then
+					newScores[winner] = (newScores[winner] or 0) + 1
+				end
+		end
+	end
+	if not opponent1.score and foundScores then
+		opponent1.score = newScores[1] or 0
+	end
+	if not opponent2.score and foundScores then
+		opponent2.score = newScores[2] or 0
+	end
+	match.opponent1 = opponent1
+	match.opponent2 = opponent2
+	return match
+end
+
+function matchFunctions.readDate(matchArgs)
+	if matchArgs.date then
+		local dateProps = MatchGroupInput.readDate(matchArgs.date)
+		dateProps.hasDate = true
+		return dateProps
+	else
+		return {
+			date = EPOCH_TIME_EXTENDED,
+			dateexact = false,
+		}
+	end
+end
+
+function matchFunctions.getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'team'))
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_valve_tier'))
+	match.status = Logic.emptyOr(match.status, Variables.varDefault('tournamentSTATUS'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function matchFunctions.getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
+	return match
+end
+
+function matchFunctions.getMatchStatus(match)
+	if Table.includes(NOT_PLAYED_MATCH_STATUSES, match.resulttype) then
```

Comment:
match.resulttype should already be normalized, so only check against
```suggestion
	if match.resulttype == NOT_PLAYED_RESULT_TYPE then
```

with `local NOT_PLAYED_RESULT_TYPE = 'np'